swarm/src/behaviour.rs

```diff
@@ -721,3 +838,214 @@ impl Default for CloseConnection {
         CloseConnection::All
     }
 }
+
+pub enum InEvent<'a, Handler: IntoConnectionHandler> {
+    /// Informs the behaviour about a newly established connection to a peer.
+    ConnectionEstablished {
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        endpoint: &'a ConnectedPoint,
+        // TODO: Would a slice not be better?
```

Comment:
Is there a semantic difference between `None` and `Some([])`?

Reply:
Good point. I don't think there is. 99a4b64f removes the `Option` and moves to a `&[]`.